## 1. The World Bank's international debt data
<p>It's not that we humans only take debts to manage our necessities. A country may also take debt to manage its economy. For example, infrastructure spending is one costly ingredient required for a country's citizens to lead comfortable lives. <a href="https://www.worldbank.org">The World Bank</a> is the organization that provides debt to countries.</p>
<p>In this notebook, we are going to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. We are going to find the answers to questions like: </p>
<ul>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
</ul>
<p><img src="https://assets.datacamp.com/production/project_754/img/image.jpg" alt></p>
<p>The first line of code connects us to the <code>international_debt</code> database where the table <code>international_debt</code> is residing. Let's first <code>SELECT</code> <em>all</em> of the columns from the <code>international_debt</code> table. Also, we'll limit the output to the first ten rows to keep the output clean.</p>

In [92]:
%%sql
postgresql:///international_debt
SELECT *
FROM international_debt
LIMIT 10;
    

10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724.300000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847181.900000006')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894453.700000003'))]

## 2. Finding the number of distinct countries
<p>From the first ten rows, we can see the amount of debt owed by <em>Afghanistan</em> in the different debt indicators. But we do not know the number of different countries we have on the table. There are repetitions in the country names because a country is most likely to have debt in more than one debt indicator. </p>
<p>Without a count of unique countries, we will not be able to perform our statistical analyses holistically. In this section, we are going to extract the number of unique countries present in the table. </p>

In [94]:
%%sql
SELECT COUNT(DISTINCT country_name) AS total_distinct_countries
FROM international_debt

 * postgresql:///international_debt
1 rows affected.


[(124,)]

## 3. Finding out the distinct debt indicators
<p>We can see there are a total of 124 countries present on the table. As we saw in the first section, there is a column called <code>indicator_name</code> that briefly specifies the purpose of taking the debt. Just beside that column, there is another column called <code>indicator_code</code> which symbolizes the category of these debts. Knowing about these various debt indicators will help us to understand the areas in which a country can possibly be indebted to. </p>

In [96]:
%%sql
SELECT COUNT(DISTINCT indicator_name) AS distinct_debt_indicators
FROM international_debt
ORDER BY distinct_debt_indicators;

 * postgresql:///international_debt
1 rows affected.


[(25,)]

## 4. Totaling the amount of debt owed by the countries
<p>As mentioned earlier, the financial debt of a particular country represents its economic state. But if we were to project this on an overall global scale, how will we approach it?</p>
<p>Let's switch gears from the debt indicators now and find out the total amount of debt (in USD) that is owed by the different countries. This will give us a sense of how the overall economy of the entire world is holding up.</p>

In [98]:
%%sql
SELECT ROUND(SUM(debt), 2) AS total_debt
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(Decimal('3079734487675.80'),)]

## 5. Country with the highest debt
<p>"Human beings cannot comprehend very large or very small numbers. It would be useful for us to acknowledge that fact." - <a href="https://en.wikipedia.org/wiki/Daniel_Kahneman">Daniel Kahneman</a>. That is more than <em>3 million <strong>million</strong></em> USD, an amount which is really hard for us to fathom. </p>
<p>Now that we have the exact total of the amounts of debt owed by several countries, let's now find out the country that owns the highest amount of debt along with the amount. <strong>Note</strong> that this debt is the sum of different debts owed by a country across several categories. This will help to understand more about the country in terms of its socio-economic scenarios. We can also find out the category in which the country owns its highest debt. But we will leave that for now. </p>

In [100]:
%%sql
SELECT 
    country_name,
    SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC;

 * postgresql:///international_debt
124 rows affected.


[('China', Decimal('285793494734.200001568')),
 ('Brazil', Decimal('280623966140.800007581')),
 ('South Asia', Decimal('247608723990.600003211')),
 ('Least developed countries: UN classification', Decimal('212880992791.900000988')),
 ('Russian Federation', Decimal('191289057259.200001943')),
 ('IDA only', Decimal('179048127207.299999298')),
 ('Turkey', Decimal('151125758035.300003616')),
 ('India', Decimal('133627060958.399997148')),
 ('Mexico', Decimal('124596786217.300001668')),
 ('Indonesia', Decimal('113435696693.499999149')),
 ('Cameroon', Decimal('86491206346.599999659')),
 ('Angola', Decimal('71368842500.100001399')),
 ('Kazakhstan', Decimal('70159942693.900001356')),
 ('Egypt, Arab Rep.', Decimal('62077727756.899998860')),
 ('Vietnam', Decimal('45851299896.100000276')),
 ('Colombia', Decimal('45430117605.200000882')),
 ('Pakistan', Decimal('45139315398.499999910')),
 ('Romania', Decimal('42813979498.199999923')),
 ('South Africa', Decimal('36703940742.499998593')),
 ('Venezuela, RB', Decimal('36048260107.700000791')),
 ('Bangladesh', Decimal('35045492840.499999572')),
 ('Lebanon', Decimal('29697872618.800000441')),
 ('Ukraine', Decimal('28490304100.300000656')),
 ('Sri Lanka', Decimal('26764495009.400000141')),
 ('Philippines', Decimal('25151706784.999999640')),
 ('Peru', Decimal('24091780491.899999879')),
 ('Belarus', Decimal('23273359790.899999251')),
 ('Kenya', Decimal('22622406389.799999776')),
 ('Morocco', Decimal('21492166822.400000074')),
 ('Ecuador', Decimal('17510981502.599999755')),
 ('Thailand', Decimal('17236647553.900000043')),
 ('Serbia', Decimal('14740351568.700000033')),
 ('Georgia', Decimal('14379874754.500000303')),
 ('Tunisia', Decimal('14223413457.600000084')),
 ('Nigeria', Decimal('14149905240.899999918')),
 ('Ethiopia', Decimal('13718630481.300000012')),
 ('Ghana', Decimal('11462311995.199999901')),
 ('Dominican Republic', Decimal('10351687705.300000106')),
 ('Mongolia', Decimal('9847249858.899999768')),
 ('Uzbekistan', Decimal('9532699072.699999864')),
 ('Zambia', Decimal('9430506984.900000065')),
 ('El Salvador', Decimal('9411612674.600000233')),
 ('Papua New Guinea', Decimal('8784109797.200000182')),
 ('Azerbaijan', Decimal('8753732947.200000224')),
 ('Costa Rica', Decimal('8616331219.900000035')),
 ('Jordan', Decimal('8472720622.900000062')),
 ('Lao PDR', Decimal('8356392647.199999973')),
 ('Guatemala', Decimal('7932838900.700000417')),
 ('Bolivia', Decimal('7889146776.799999981')),
 ('Tanzania', Decimal('7798052922.500000175')),
 ("Cote d'Ivoire", Decimal('7141829014.799999847')),
 ('Senegal', Decimal('7119017873.999999940')),
 ('Bosnia and Herzegovina', Decimal('6685436904.600000101')),
 ('Bulgaria', Decimal('6369101653.900000139')),
 ('Uganda', Decimal('6139129245.399999968')),
 ('Myanmar', Decimal('6130613635.299999843')),
 ('Paraguay', Decimal('5774787846.799999941')),
 ('Nicaragua', Decimal('5399238587.099999943')),
 ('Mauritius', Decimal('4787131190.800000070')),
 ('Kyrgyz Republic', Decimal('4656980235.999999940')),
 ('Mozambique', Decimal('4608729187.499999977')),
 ('Cambodia', Decimal('4229626566.599999795')),
 ('Jamaica', Decimal('3981335937.899999976')),
 ('Sudan', Decimal('3835096534.199999988')),
 ('Armenia', Decimal('3834876451.500000035')),
 ('Macedonia, FYR', Decimal('3502007528.500000066')),
 ('Nepal', Decimal('3405713007.199999991')),
 ('Moldova', Decimal('3338411353.800000059')),
 ('Albania', Decimal('3217067103.800000042')),
 ('Gabon', Decimal('3187705162.200000028')),
 ('Mauritania', Decimal('3184696802.099999934')),
 ('Montenegro', Decimal('2992522191.999999998')),
 ('Mali', Decimal('2604524204.399999909')),
 ('Yemen, Rep.', Decimal('2552698189.200000010')),
 ('Honduras', Decimal('2543918981.500000007')),
 ('Benin', Decimal('2535179229.299999979')),
 ('Niger', Decimal('2478419343.000000121')),
 ('Burkina Faso', Decimal('2419206260.700000108')),
 ('Congo, Dem. Rep.', Decimal('2239101510.600000068')),
 ('Syrian Arab Republic', Decimal('1840057923.000000021')),
 ('Madagasca

## 6. Average amount of debt across indicators
<p>So, it was <em>China</em>. A more in-depth breakdown of China's debts can be found <a href="https://datatopics.worldbank.org/debt/ids/country/CHN">here</a>. </p>
<p>We now have a brief overview of the dataset and a few of its summary statistics. We already have an idea of the different debt indicators in which the countries owe their debts. We can dig even further to find out on an average how much debt a country owes? This will give us a better sense of the distribution of the amount of debt across different indicators.</p>

In [102]:
%%sql 
SELECT 
    indicator_name,
    indicator_code,
    AVG(debt) AS Avg_To_indicator
FROM international_debt
GROUP BY indicator_name, indicator_code
ORDER BY Avg_To_indicator DESC;

 * postgresql:///international_debt
25 rows affected.


[('Principal repayments on external debt, long-term (AMT, current US$)', 'DT.AMT.DLXF.CD', Decimal('5904868401.499193612')),
 ('Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', 'DT.AMT.DPNG.CD', Decimal('5161194333.812658349')),
 ('Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('2152041216.890243888')),
 ('PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('1958983452.859836046')),
 ('PPG, private creditors (AMT, current US$)', 'DT.AMT.PRVT.CD', Decimal('1803694101.963265321')),
 ('Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('1644024067.650806481')),
 ('PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('1223139290.398230108')),
 ('Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', 'DT.INT.DPNG.CD', Decimal('1220410844.421518983')),
 ('PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('1191187963.083064523')),
 ('PPG, bonds (AMT, current US$)', 'DT.AMT.PBND.CD', Decimal('1082623947.653623188')),
 ('PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('839843678.615833335')),
 ('PPG, bonds (INT, current US$)', 'DT.INT.PBND.CD', Decimal('804733376.598550709')),
 ('PPG, other private creditors (AMT, current US$)', 'DT.AMT.PROP.CD', Decimal('746888800.477777779')),
 ('PPG, commercial banks (AMT, current US$)', 'DT.AMT.PCBK.CD', Decimal('734868742.601190462')),
 ('PPG, private creditors (INT, current US$)', 'DT.INT.PRVT.CD', Decimal('719740180.088775506')),
 ('PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('712619634.659836064')),
 ('PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('490062193.498387090')),
 ('PPG, private creditors (DIS, current US$)', 'DT.DIS.PRVT.CD', Decimal('311323264.690566033')),
 ('PPG, official creditors (INT, current US$)', 'DT.INT.OFFT.CD', Decimal('297677338.957258067')),
 ('PPG, commercial banks (DIS, current US$)', 'DT.DIS.PCBK.CD', Decimal('293305195.594117648')),
 ('PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('164093285.609016397')),
 ('PPG, commercial banks (INT, current US$)', 'DT.INT.PCBK.CD', Decimal('156647613.055952384')),
 ('PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('136230719.245161288')),
 ('PPG, other private creditors (DIS, current US$)', 'DT.DIS.PROP.CD', Decimal('81135160.699999999895')),
 ('PPG, other private creditors (INT, current US$)', 'DT.INT.PROP.CD', Decimal('34250651.235185185333'))]

## 7. The highest amount of principal repayments
<p>We can see that the indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital. More information about this category can be found <a href="https://datacatalog.worldbank.org/principal-repayments-external-debt-long-term-amt-current-us-0">here</a>. </p>
<p>An interesting observation in the above finding is that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.</p>
<p>We can investigate this a bit more so as to find out which country owes the highest amount of debt in the category of long term debts (<code>DT.AMT.DLXF.CD</code>). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically. </p>

In [104]:
%%sql
WITH CTE AS 
(
    SELECT country_name,
           ROW_NUMBER() OVER(ORDER BY debt DESC) AS RK
    FROM international_debt
    WHERE indicator_code = 'DT.AMT.DLXF.CD'
)

SELECT country_name
FROM CTE
WHERE RK = 1;

 * postgresql:///international_debt
1 rows affected.


[('China',)]

## 8. The most common debt indicator
<p>China has the highest amount of debt in the long-term debt (<code>DT.AMT.DLXF.CD</code>) category. This is verified by <a href="https://data.worldbank.org/indicator/DT.AMT.DLXF.CD?end=2018&most_recent_value_desc=true">The World Bank</a>. It is often a good idea to verify our analyses like this since it validates that our investigations are correct. </p>
<p>We saw that long-term debt is the topmost category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? Let's find that out. </p>

In [106]:
%%sql 
SELECT indicator_code,
       COUNT(indicator_code) AS common_debt_indicator
FROM international_debt
GROUP BY indicator_code
ORDER BY common_debt_indicator DESC
LIMIT 1;

 * postgresql:///international_debt
1 rows affected.


[('DT.INT.DLXF.CD', 124)]

## 9. Other viable debt issues and conclusion
<p>There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator <code>DT.AMT.DLXF.CD</code> is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue. But that is not the end of the story, but just a part of the story.</p>
<p>Let's change tracks from <code>debt_indicator</code>s now and focus on the amount of debt again. Let's find out the maximum amount of debt that each country has. With this, we will be in a position to identify the other plausible economic issues a country might be going through.</p>
<p>In this notebook, we took a look at debt owed by countries across the globe. We extracted a few summary statistics from the data and unraveled some interesting facts and figures. We also validated our findings to make sure the investigations are correct.</p>

In [108]:
%%sql
SELECT country_name,
       MAX(debt) AS Max_debt
FROM international_debt
GROUP BY country_name
ORDER BY Max_debt DESC
LIMIT 10;

 * postgresql:///international_debt
10 rows affected.


[('China', Decimal('96218620835.699996948')),
 ('Brazil', Decimal('90041840304.100006104')),
 ('Russian Federation', Decimal('66589761833.5')),
 ('Turkey', Decimal('51555031005.800003052')),
 ('South Asia', Decimal('48756295898.199996948')),
 ('Least developed countries: UN classification', Decimal('40160766261.599998474')),
 ('IDA only', Decimal('34531188113.199996948')),
 ('India', Decimal('31923507000.799999237')),
 ('Indonesia', Decimal('30916112653.799999237')),
 ('Kazakhstan', Decimal('27482093686.400001526'))]